pkgs/development/python-modules/awscrt/default.nix

```
@@ -0,0 +1,30 @@
+{ lib, buildPythonPackage, fetchPypi, cmake, gcc10, perl }:
+
+buildPythonPackage rec {
+  version = "0.11.13";
+  pname = "awscrt";
+
+  propagateBuildInputs = [ cmake gcc10 perl ];
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-G/bf2AzWp8AHL4of0zfX3jIYyTtmTLBIC2ZKiMi19c0=";
+  };
+
+  postPatch = ''
+    export PATH="${cmake}/bin:$PATH"
+    export PATH="${gcc10}/bin:$PATH"
+    export PATH="${perl}/bin:$PATH"
+  '';
```

```suggestion
  nativeBuildInputs = [ cmake gcc10 perl ];
```

`nativeBuildInputs` is for packages that will be executed at build time, so it has this effect.

---

A version of gcc is part of the default stdenv on linux.
Which compiler do we need on darwin? That stdenv comes with clang, which may be ok.

---

Not sure, but can give this a run on a mac tonight.

---

It works with clang, but it needed `(with darwin.apple_sdk.frameworks; [ Security ]);`